### PR TITLE
Remove rule logging and related code

### LINKED ISSUE #2655: Deprecated config options `RULE_LOG_DIR` and `RULE_LOG_IGNORE`
The config options `iris.config.RULE_LOG_IGNORE` and `iris.config.RULE_LOG_DIR` were deprecated Iris 1.10

[RULE_LOG_DIR](https://github.com/SciTools/iris/blob/master/lib/iris/config.py#L50)
[RULE_LOG_IGNORE](https://github.com/SciTools/iris/blob/master/lib/iris/config.py#L58)

### PR DESCRIPTION
**Pull request by LukeC92**
This is a continuation of @dkillick 's Pull Request #2697 . As well as removing rule logging config items and all rule logging functionality from `fileformats/rules.py` I also removed `test_verbose_fileformat_rules_logging.py` which now seems redundant, and a referece to rule logs in  `pp.save_pairs_from_cube` .

Closes #2655

### CODE REVIEW

File: lib/iris/config.py [hunk 1]
```diff
@@ -148,13 +134,6 @@ def get_dir_option(section, option, default=None):
 # Logging options
 _LOGGING_SECTION = 'Logging'
```
Comment (corinnebosley) [inline lib/iris/config.py]: @LukeC92 @dkillick @pelson @pp-mo @djkirkham 
I know that this entire set of logging options is not deprecated, but while we are working towards Iris 2.0 and have also just implemented Python logging (with a view to further development), I really think we should take a long hard look at whether we still need this and should absolutely remove it if we don't.

Comment (pelson) [reply lib/iris/config.py]: Yep. ``import_logger`` can be removed long-term.
Definitely a separate PR, but I'd probably support it happening for 2.0 without a deprecation warning. We will have to document the breaking change though (and think about the impact for people who have a logging section in their config - would it now break if we removed it, or just be ignored silently?).

All questions to be answered outside of this PR.

**Comment by pelson** [comment]
My favourite kind of PR - one that makes the codebase smaller & tidier. 
Thanks @LukeC92!

**Comment by LukeC92** [comment]
I've started writing up the "What's New" for this pull request but I am not sure what the best way is to represent all of these changes.

**Pull request merged**